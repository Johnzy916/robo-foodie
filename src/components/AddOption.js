import React, { Component } from 'react';
import { connect } from 'react-redux';
import { startAddPlace } from '../actions/places';

export class AddOption extends Component {
    state = {
        error: undefined
    }
    

  handleAddPlace = (e) => {
    e.preventDefault();
    const place = e.target.elements.place.value.trim();
    let error;
        if (!place) {
            error = 'Please enter something you\'d like to eat';
        } else if (this.props.places.map(value => value.place).indexOf(place) > -1) {
            error = 'This option already exists!';
            e.target.elements.place.value = "";
        }
        this.setState(() => ({ error }));
        if (place && !error) {
            this.props.startAddPlace(place);
            e.target.elements.place.value = "";
        }
  };

  render() {
    return (
      <div>
        {this.state.error &&
          <p className="add-place-error">
            {this.state.error}
          </p>}
        <form
          className="add-place"
          onSubmit={this.handleAddPlace}
        >
          <input 
            className="add-place__input" 
            type="text" 
            name="place" 
            placeholder="Restaurant, meal, ingredient..."
          />
          <button className="btn btn--primary btn--shine">Add Option</button>
        </form>
      </div>
    );
  }
}

const mapStateToProps = state => ({
   places: state.places 
});

const mapDispatchToProps = dispatch => ({
   startAddPlace: place => dispatch(startAddPlace(place)) 
});

export default connect(mapStateToProps, mapDispatchToProps)(AddOption);