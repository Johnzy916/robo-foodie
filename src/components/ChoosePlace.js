import React from 'react';
import { connect } from 'react-redux';
import { selectOption } from '../actions/option';
import { showModal } from '../actions/modal';

export const ChoosePlace = ({ 
        places,
        selectOption, 
        showModal, 
        isFirstTimeUser, 
        firstTimeModal }) => {
    
    const handleMakeDecision = () => {
        const randomNum = Math.floor(Math.random() * places.length);
        const option = places[randomNum];
        selectOption(option);
        showModal(option);
    }

    return (
      <div>
        <button 
          disabled={places.length < 2}
          onClick={handleMakeDecision}
          className="btn btn--secondary btn--shine big-button"
        >
          What are we eating?
        </button>
      </div>
    );
};

const mapStateToProps = state => ({
   places: state.places,
});

const mapDispatchToProps = dispatch => ({
    selectOption: option => dispatch(selectOption(option)),
    showModal: option => dispatch(showModal({
      contentLabel: 'Confirm Selection',
      header: `It looks like we'll be eating:`,
      subtitle: `${option.place}`,
      confirmText: 'Great!',
      confirmType: 'confirm',
    }))
})

export default connect(mapStateToProps, mapDispatchToProps)(ChoosePlace);