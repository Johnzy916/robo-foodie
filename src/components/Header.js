import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom'
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header className="header">
      <div className="content-container">
        <nav className="header__content">
          <Link
            className="header__title"
            to="/dashboard">
              <h1>Robo-Foodie</h1>
          </Link>
          <button
            className="btn btn--primary btn--link"
            onClick={startLogout}>
              Logout
          </button>
        </nav>
      </div>
  </header>
);

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);