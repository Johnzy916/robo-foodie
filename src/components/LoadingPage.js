import React from 'react';

const LoadingPage = () => (
    <div className="loader">
        <img className="loader__image" src="/images/loader-dark.gif" />
    </div>
);

export default LoadingPage;