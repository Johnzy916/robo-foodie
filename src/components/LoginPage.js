import React, { Component } from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Robo-Foodie</h1>
            <p className="box-layout__tagline">Take the back-and-forth out of what you're eating!</p>
            <p className="box-layout__sub-tagline">Sign in to save and access your places</p>
            <button 
                className="btn btn--google btn--shine box-layout__button"
                onClick={() => startLogin('google')}>
                    <img
                        className="box-layout__button-logo"
                        src="/images/brands/google-logo.png" alt="google logo" 
                    />
                    Login with Google
            </button>
            <button 
                className="btn btn--github btn--shine box-layout__button"
                onClick={() => startLogin('github')}>
                    <img
                        className="box-layout__button-logo"
                        src="/images/brands/github-logo.png" alt="google logo" 
                    />
                    Login with Github
            </button>
            <button 
                className="btn btn--facebook btn--shine box-layout__button"
                onClick={() => startLogin('facebook')}>
                    <img
                        className="box-layout__button-logo"
                        src="/images/brands/facebook-logo.png" alt="google logo" 
                    />
                    Login with Facebook
            </button>
        </div>
    </div>
);

const mapDispatchToProps = dispatch => ({
    startLogin: (type) => dispatch(startLogin(type))
});

export default connect(undefined, mapDispatchToProps)(LoginPage);