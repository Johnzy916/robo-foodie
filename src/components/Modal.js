import React from 'react';
import ReactModal from 'react-modal';
import { connect } from 'react-redux';
import {  hideModal, modalConfirm } from '../actions/modal';
import { history } from '../routers/AppRouter';

if (process.env.NODE_ENV !== 'test') ReactModal.setAppElement('#app');

export const Modal = ({ modalVisible, modalProps, modalType, cancelModal, confirmModal }) => (
    <ReactModal
        isOpen={modalVisible}
        contentLabel={modalProps.contentLabel}
        onRequestClose={cancelModal}
        closeTimeoutMS={200}
        className="react-modal"
    >
        <h3 className="react-modal__body">
            {modalProps.header}
        </h3>
        { modalProps.subtitle &&
        <p className="react-modal__info">
            {modalProps.subtitle}
        </p>
        }
        { modalProps.cancelText &&
            <button
                className="btn btn--tertiary btn--shine react-modal__btn"
                onClick={() => {
                        cancelModal();
                        if (modalProps.cancelAuth) { 
                            modalProps.cancelAuth('User chose not to connect accounts');
                        };
                    }
                }
            >
                {modalProps.cancelText}
            </button>
        }
        <button 
            className="btn btn--primary btn--shine react-modal__btn"
            onClick={cancelModal}
        >
            {modalProps.confirmText}
        </button>
    </ReactModal>
);

const mapStateToProps = state => ({
    modalType: state.modal.modalType,
    modalVisible: state.modal.modalVisible,
    modalProps: state.modal.modalProps
});

const mapDispatchToProps = (dispatch, props) => ({
    cancelModal: () => {
        dispatch(hideModal())
    },
    confirmModal: (confirmType, id) => {
        dispatch(hideModal());
        dispatch(modalConfirm(confirmType, id));
        history.push('/');
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(Modal);