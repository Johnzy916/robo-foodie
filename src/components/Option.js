import React from 'react';
import { connect } from 'react-redux';
import { startRemovePlace } from '../actions/places';

export const Option = props => (
  <div className="option">
    <p className="option__text">{props.optionText}</p>
    <button
      className="btn btn--tertiary--dark btn--link option__btn"
      onClick={() => props.startRemovePlace(props.id)}
    >
      remove
    </button>
  </div>
);

const mapDispatchToProps = dispatch => ({
   startRemovePlace: id => dispatch(startRemovePlace(id)) 
});

export default connect(undefined, mapDispatchToProps)(Option);