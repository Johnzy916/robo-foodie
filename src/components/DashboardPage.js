import React, { Component } from 'react';
import { connect } from 'react-redux';
import AddOption from './AddOption';
import Options from './Options';
import ChoosePlace from './ChoosePlace';
import { isFirstTimeUser } from '../actions/auth';

export const DashboardPage = ({ isFirstTimeUser, closeFirstTimeAddon }) => (
  <div>
    <div className="page-header">
      <div className="content-container">
        <div className="page-header__content">
          <ChoosePlace />
          { isFirstTimeUser &&
            <div className="addon">
              <h3 className="addon__title">
                I see it's your first time here!
              </h3>
              <p className="addon__text">
                Add the places or foods you'd like to eat and press the button to let Robo decide!
              </p>
              <button
                className="addon__close-btn"
                onClick={closeFirstTimeAddon}
              >
                &times;
              </button>
            </div>
          }
        </div>
      </div>
    </div>
    <div className="content-container">
      <div className="widget">
        <Options />
        <AddOption />
      </div>
    </div>
  </div>
);

const mapStateToProps = state => ({
  isFirstTimeUser: state.auth.isFirstTimeUser
});

const mapDispatchToProps = dispatch => ({
  closeFirstTimeAddon: () => dispatch(isFirstTimeUser(false))
});


export default connect(mapStateToProps, mapDispatchToProps)(DashboardPage);
