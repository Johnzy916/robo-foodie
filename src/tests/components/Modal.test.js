import React from 'react';
import { shallow } from 'enzyme';
import { Modal } from '../../components/Modal';

let wrapper, cancelModal, confirmModal;

beforeEach(() => {
  cancelModal = jest.fn();
  confirmModal = jest.fn();
  wrapper = shallow(<Modal
    modalVisible={true}
    modalProps={{
        contentLabel: 'Delete expense',
        header: 'Are you sure you want to delete this expense?',
        cancelText: 'Keep expense',
        confirmText: 'Yes, delete',
        confirmType: 'remove',
        id: '1'
    }}
    modalType='remove'
    cancelModal={cancelModal}
    confirmModal={confirmModal}
  />)
});

test('should render modal correctly', () => {
    expect(wrapper).toMatchSnapshot();
})

test('should handle cancel button', () => {
  wrapper.find('button').at(0).simulate('click');
  expect(cancelModal).toHaveBeenCalled();
});