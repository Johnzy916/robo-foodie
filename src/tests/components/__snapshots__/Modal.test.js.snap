// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render modal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="react-modal"
  closeTimeoutMS={200}
  contentLabel="Delete expense"
  isOpen={true}
  onRequestClose={[MockFunction]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="react-modal__body"
  >
    Are you sure you want to delete this expense?
  </h3>
  <button
    className="btn btn--tertiary btn--shine react-modal__btn"
    onClick={[Function]}
  >
    Keep expense
  </button>
  <button
    className="btn btn--primary btn--shine react-modal__btn"
    onClick={[MockFunction]}
  >
    Yes, delete
  </button>
</Modal>
`;
